src/cpu/vendor_model_name/vendor_id.cpp

```diff
@@ -27,3 +30,22 @@ std::string iware::cpu::vendor_id() {
 
 	return name;
 }
+#elif INFOWARE_APPLE && INFOWARE_ARM
+#include <sys/sysctl.h>
+
+std::string iware::cpu::vendor_id() {
+      int cmd[2] = { CTL_KERN, HW_MACHINE };
+
+      size_t buf_len = 0;
+      std::string name;
+      int ret = sysctl(cmd, 2, nullptr, &buf_len, nullptr, 0);
+
+      if (ret < 0) {
+            return "Unknown";     
+      }
+      name.resize(buf_len);
+      ret = sysctl(cmd, 2, &name[0], &buf_len, nullptr, 0);
+      
+      return ret < 0? "Unknown" : name;
```

Comment:
Change the `infoware/detail/sysctl.hpp` API to return `std::string`, have this be `return iware::detail::sysctl(CTL_KERN, HW_MACHINE);`, and replace `vendor_model_name_darwin.cpp` with
```cpp
#ifdef __APPLE__

#include "infoware/cpu.hpp"
#include "infoware/detail/sysctl.hpp"

std::string iware::cpu::vendor() {
	return iware::detail::sysctl("machdep.cpu.vendor");
}

std::string iware::cpu::model_name() {
	return iware::detail::sysctl("machdep.cpu.brand_string");
}

#endif
```
and replace the function body in `os_info_darwin.cpp` with
```cpp
	auto ctl_osrelease_data = iware::detail::sysctl("kern.osrelease");
	char buf{};
	char* osrelease_marker = ctl_osrelease_data.empty() ? &buf : ctl_osrelease_data.data();

	const auto ctl_osrevision_data = iware::detail::sysctl("kern.osrevision");
	unsigned int build_number{};
	if(!ctl_osrevision_data.empty()) {
		const auto osrevision_data = iware::detail::deconstruct_sysctl_int(ctl_osrevision_data);
		if(osrevision_data.first)
			build_number = osrevision_data.second;
	}

	return {iware::detail::sysctl("kern.ostype"),                                                  //
	        iware::detail::sysctl("kern.version"),                                                 //
	        static_cast<unsigned int>(std::strtoul(osrelease_marker, &osrelease_marker, 10)),      //
	        static_cast<unsigned int>(std::strtoul(osrelease_marker + 1, &osrelease_marker, 10)),  //
	        static_cast<unsigned int>(std::strtoul(osrelease_marker + 1, nullptr, 10)),            //
	        build_number};
```